### PR TITLE
Fix ZeroDivisionError in linkx_homophily

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Summary

`linkx_homophily` crashes with a `ZeroDivisionError` in two edge cases:

1. **Zero in-degree class** -- When all nodes belonging to a particular class have zero in-degree (isolated nodes or no incoming edges), the per-class ratio `same_class_deg_k / deg_k` divides by zero. This is fixed by skipping classes with zero total degree, since there are no neighbours and the same-class fraction is undefined. The `max(0, ...)` clamp means the only sensible default contribution is 0.

2. **Single node class** -- When every node has the same label (`num_classes == 1`), the final normalization `value / (num_classes - 1)` divides by zero. The LINKX homophily metric is mathematically undefined in this case, so a clear `ValueError` is now raised.

### Reproducer

```python
import dgl, torch

# Case 1: class with zero in-degree triggers ZeroDivisionError
g = dgl.graph(([0], [1]))
y = torch.tensor([0, 0, 1])  # node 2 is isolated, class 1 has deg_k=0
dgl.linkx_homophily(g, y)  # ZeroDivisionError

# Case 2: single class triggers ZeroDivisionError
g = dgl.graph(([0, 1], [1, 0]))
y = torch.tensor([0, 0])
dgl.linkx_homophily(g, y)  # ZeroDivisionError
```

## Test plan

- [ ] Existing `test_linkx_homophily` tests continue to pass (the fix only adds guards for edge cases)
- [ ] Case 1 reproducer no longer crashes, returns `0.0`
- [ ] Case 2 reproducer raises a clear `ValueError` instead of a cryptic `ZeroDivisionError`

### CODE REVIEW

**Comment by Mr-Neutr0n** [comment]
Closing this one to tidy up my open pull requests.

It's been open around five months with no review activity, and I'd rather withdraw it than leave it sitting in your queue. Nothing needed from you, and no hard feelings at all.

If the fix is still wanted, this can be reopened, or I'm happy to redo it properly against current main. Apologies for the noise.
